plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java

```diff
@@ -104,12 +110,24 @@ public static void addMavenLibraries(IProject project, List<Library> libraries,
       }
       subMonitor.worked(1);
     }
-    javaProject.setRawClasspath(newRawClasspath.toArray(new IClasspathEntry[0]), subMonitor);
+    
+    if (addToClasspath) {
+      javaProject.setRawClasspath(newRawClasspath.toArray(new IClasspathEntry[0]), subMonitor);
+    }
     
     runContainerResolverJob(javaProject);
```

Comment:
Currently, `LibraryClasspathContainerResolverService.resolveAll()` calls `resolveContainer()`, which in turn calls `JavaCore.setClasspathContainer()`, so this may not be that simple. (It's not clear to me either `getNewContainers()` may do some overlapping work about containers.) @akerekes probably has a better idea.